internal/pkg/cli/pipeline_deploy.go

```diff
@@ -148,7 +151,11 @@ func newDeployPipelineOpts(vars deployPipelineVars) (*deployPipelineOpts, error)
 		wsAppName: wsAppName,
 		svcBuffer: &bytes.Buffer{},
 		jobBuffer: &bytes.Buffer{},
-	}, nil
+	}
+	opts.configureDeployedPipelineLister = func() deployedPipelineLister {
+		return deploy.NewPipelineStore(opts.appName, rg.New(defaultSession))
```

Comment:
nit: I think this is a little subtle, what do you think of dropping a comment saying:
```suggestion
		// Initialize the client only after the appName is asked.
		return deploy.NewPipelineStore(opts.appName, rg.New(defaultSession))
```